src/k8s/api/v1/cluster_config.go

```diff
@@ -65,6 +66,11 @@ type NetworkConfig struct {
 	Enabled *bool `json:"enabled,omitempty" yaml:"enabled"`
 }
 
+type APIServerConfig struct {
+	Datastore    string `json:"datastore,omitempty" yaml:"datastore"`
+	DatastoreURL string `json:"datastore-url,omitempty" yaml:"datastore-url"`
+}
+
```

Comment:
Note that the `UserFacingClusterConfig` struct (message, in RPC terms) is used in multiple places, like for example `k8s get/set`. We do not want this as it causes confusion:

- we can now pass such a struct to `k8s set`, but it will be silently ignored. 
- `k8s get` will give me back an APIServerConfig that will always be empty.

Please instead only add the field to `type ClusterStatus struct` instead.